### PR TITLE
use the existing utilerrors.AggregateGoroutines func to collect each cluster's replicas concurrently

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Improve code reuse

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/estimator/client/accurate.go [hunk 1]
```diff
@@ -127,22 +127,18 @@ func getClusterReplicasConcurrently(parentCtx context.Context, clusters []string
...
-	var wg sync.WaitGroup
-	errChan := make(chan error, len(clusters))
-	for i := range clusters {
-		wg.Add(1)
-		go func(idx int, cluster string) {
-			defer wg.Done()
+	clusterReplicas := make([]workv1alpha2.TargetCluster, len(clusters))
+	funcs := make([]func() error, len(clusters))
+	for i, cluster := range clusters {
+		i, cluster := i, cluster
```
Comment (Garrybest) [inline pkg/estimator/client/accurate.go:133]: I'd like to change the name in order to prevent confusion.

Comment (Garrybest) [inline pkg/estimator/client/accurate.go:133]: I'd like to change the name in order to prevent confusion.

**Review by Garrybest — COMMENTED** [review]
Generally LGTM.

File: pkg/estimator/client/accurate.go (hunk 1 above)
Comment (Garrybest) [reply pkg/estimator/client/accurate.go:133]: Rename the variables `i, cluster` in inner loop may be better.

Comment (carlory) [reply pkg/estimator/client/accurate.go]: @Garrybest updated

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**